daemon/container.go

```diff
@@ -296,10 +296,17 @@ func (container *Container) Start() (err error) {
 		return err
 	}
 
+	if !(container.hostConfig.IpcMode.IsContainer() || container.hostConfig.IpcMode.IsHost()) {
```

Comment:
Would be easier for me to read `!X && !Y` than `!(X || Y)` but it's my personal taste, feel free to disagree